tests/integration/fee-estimation.test.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  parachainUniversalLocation,
+  Registry,
+  relaychainUniversalLocation,
+} from '@open-xcm-tools/simple-xcm';
+import {universalLocation, location} from '@open-xcm-tools/xcm-util';
+import {describe, expect, test} from 'vitest';
+
+describe('fee estimation tests', async () => {
+  const BDK_URL = process.env.BDK_BALANCER!.replace('http', 'ws');
+
+  const registry = new Registry()
+    .addChain({
+      identity: {
+        name: 'Polkadot',
+        universalLocation: relaychainUniversalLocation('westend'),
+      },
+      endpoints: [`${BDK_URL}/relay/`],
+    })
+    .addChain({
+      identity: {
+        name: 'AssetHubA',
+        universalLocation: parachainUniversalLocation('westend', 2001n),
+      },
+      endpoints: [`${BDK_URL}/relay-assethubA/`],
+    })
+    .addChain({
+      identity: {
+        name: 'AssetHubB',
+        universalLocation: parachainUniversalLocation('westend', 2002n),
+      },
+      endpoints: [`${BDK_URL}/relay-assethubB/`],
+    })
+    .addChain({
+      identity: {
+        name: 'AssetHubC',
+        universalLocation: parachainUniversalLocation('westend', 2003n),
+      },
+      endpoints: [`${BDK_URL}/relay-assethubC/`],
+    })
+    .addCurrency({
+      symbol: 'USDT',
+      decimals: 6,
+      universalLocation: universalLocation('westend', [
+        {parachain: 2001n},
+        {palletInstance: 50n},
+        {generalIndex: 1984n},
+      ]),
+    })
+    .addRelativeLocation(
+      'Alice',
+      location(0n, [
+        {
+          accountId32: {
+            id: '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY',
+          },
+        },
+      ]),
+    );
+
+  await registry.addNativeCurrency('Polkadot');
+  const xcmAssetHubA = await registry.connectXcm('AssetHubA');
+  const xcmAssetHubB = await registry.connectXcm('AssetHubB');
+  const xcmAssetHubC = await registry.connectXcm('AssetHubC');
+
+  test('correct composeTransfer: A -> B', async () => {
```

Comment:
Please add tests that use the estimator directly. Namely, the `tryEstimateExtrinsicFees`. And use `chainXcmFeesLogger` to check that we estimated fees on remote chains